public/app/features/profile/SignupCtrl.tsx

```diff
@@ -0,0 +1,112 @@
+import React, { PureComponent } from 'react';
+import { getBackendSrv } from '@grafana/runtime';
+import { updateLocation } from 'app/core/actions';
+import { connect } from 'react-redux';
+import { StoreState } from 'app/types';
+import { hot } from 'react-hot-loader';
+
+interface SignupProps {
+  children(arg0: ChildProps): JSX.Element;
+  routeParams?: any;
+  updateLocation: typeof updateLocation;
+}
+
+export interface SignupFormModel {
+  email: string;
+  username?: string;
+  password: string;
+  orgName: string;
+  code?: string;
+  name?: string;
+}
+
+interface ChildProps {
+  onSubmit(obj: SignupFormModel): void;
+  verifyEmailEnabled: boolean;
+  autoAssignOrg: boolean;
+  defaultValues: SignupCtrlState;
+}
+
+export interface SignupCtrlState {
+  autoAssignOrg: boolean;
+  verifyEmailEnabled: boolean;
+  orgName: string;
+  email: string;
+  username: string;
+  code?: string;
+  password?: string;
+}
+
+export class SignupCtrl extends PureComponent<SignupProps, SignupCtrlState> {
+  private defaultValues: SignupCtrlState;
+  constructor(props: SignupProps) {
+    super(props);
+
+    //Set initial values from url via props.routeParams
+    this.state = this.defaultValues = {
+      verifyEmailEnabled: false,
+      autoAssignOrg: true,
+      orgName: props.routeParams.email,
+      email: props.routeParams.email,
+      username: props.routeParams.email,
+      code: props.routeParams.code,
+    };
+  }
+
+  componentDidMount() {
+    getBackendSrv()
+      .get('/api/user/signup/options')
```

Comment:
Instead of this call, is there something that block us from adding these options to boot config @marefr? I would rather avoid retrieving these options like this.

Reply:
ping @marefr ^^

Reply:
No, not really. The only thing returned is this:
https://github.com/grafana/grafana/blob/959c49f6d8f6a09f370975ba88ca082c2acd35a5/pkg/api/signup.go#L13-L19

But would suggest you remove the backend api code if it's no longer in use.

Reply:
Ok, so it's fine to put those in the Grafana boot config?

Reply:
Yeah I see no problem with that

Reply:
Are these available just by importing the config, or is there something else that needs to be done?

Reply:
Those need to be added to config in https://github.com/grafana/grafana/blob/3fabbbff4d04b14720c5f77503d170d124c0b980/pkg/setting/setting.go#L4